url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/dataStructure/AdjMatrix.java

```diff
@@ -0,0 +1,455 @@
+
+    public AdjMatrix (int nverts) {
+    	violatedNodesFlag = false;
+    	prohibitedNodes = new HashSet();
+    	allowedNodes = new HashSet();
+    	prohibitedIndexMap = new HashMap<Integer, Integer>();
+    	
+    	lastPossibleRoot = -1;
+
+    	
+	AM = new double[nverts][nverts];
+	size = nverts;
+	makeEmpty();
+    }
+
+    public void makeEmpty() {
+	for (int i = 0; i < size; i++)
+	    for (int j = 0; j < size; j++)
+		if (i == j) 
+		    AM[i][j] = 0;
+		else
+		    AM[i][j] = INFINITO;	
+    }
+
+    public int getSize() { return size; }
+    
+    private void fillMatrix(AdjList adjl) {
+    	ALNode alnode;
+    	makeEmpty();
+    	
+    	for (int i = 0; i < adjl.getNumberOfVertices(); i++) {
+    	    alnode = adjl.getList(i);
+    	    while (alnode != null) {
+    		AM[i][alnode.getVid()] = alnode.getWeight();
+    		alnode = alnode.getProx();
+    	    }
+    	}    	
+    }
+    
+    public void refillMatrix() {
+    	fillMatrix(localAdjList);
+    }
+
+    public void fillMatrixFromAdjList (AdjList adjl) {	
+	localAdjList = adjl;
+	
+	fillMatrix(localAdjList);
+    }
+
+
+    public void printAdjMatrix() {
+	int i, j;
+
+	System.out.println ("Adj Matrix:");
+	for (i = 0; i < size; i++) {
+	    System.out.print (i + ": ");
+	    for (j = 0; j < size; j++)
+		System.out.print ("  [" + AM[i][j] + "]");
+	    System.out.println();
+	}
+    }
+
+
+    ALNode insertInList (ALNode adjl, int j, double w) {
+    	ALNode p;
+    
+	p = new ALNode(j, w);
+	p.setProx(adjl);
+	adjl = p;
+    	return adjl;
+    }
+    
+    ALNode checkAndInsertInList (ALNode adjl, int j, double w) {
+    	ALNode p;
+    
+	p = new ALNode(j, w);
+	p.setProx(adjl);
+	adjl = p;
+    	return adjl;
+    }
+    
+    public double[][] getP2PCost() {
+    	return MCPmatrix;
+    }
+    
+    public int [][] getP2PPath() {
+    	return P;
+    }
+    
+    public double getMatrixCost(int x, int y, double[][] globalMatrix) {
+    	if (globalMatrix != null)
+    		return MCPmatrix2Check[x][y];
+    	else
+    		return MCPmatrix[x][y];
+    }
+    
+    private void deleteNodeTemporarily(int index, int terminalsIndex[]) {
+		prohibitedIndexMap.put(terminalsIndex[index], index);
+    	local2CheckAdjList.deleteTemporaryNode(terminalsIndex[index]);
+    	terminalsIndex[index] = -1;
+    }
+    
+    private void restoreTemporarilyDeletedNodes(int terminalsIndex[]) {
+		java.util.Set mapSet = prohibitedIndexMap.entrySet();
+		Iterator<Map.Entry<Integer, Integer>> it = mapSet.iterator();
+		while (it.hasNext()) {
+			Map.Entry entry = it.next();
+			int term = (Integer)entry.getKey();
+			int index = (Integer)entry.getValue();
+			terminalsIndex[index] = term;
+		}
+		local2CheckAdjList.restoreFromTemporaryList();
+		prohibitedIndexMap.clear();
+    }
+    
+    public AdjList getSqrtKUninformedTerminalsTreeWithRestriction(AdjMatrix globalAdjMat, int terminalsIndex[], double spannerQuant[], double sqrtT, int root[], int notAllowedToBeRoot[], 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 27. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.